spec/unit/policyfile_evaluation_spec.rb

```diff
@@ -323,6 +323,63 @@
 
     end
 
+    describe "defining attributes" do
+
+      let(:policyfile_rb) do
+        <<-EOH
+          name "policy-with-attrs"
+          run_list "foo"
+
+          # basic attribute setting:
+          default["foo"] = "bar"
+
+          # auto-vivify
+          default["abc"]["def"]["ghi"] = "xyz"
+
+          # literal data structures
+          default["baz"] = {
+            "more_nested_stuff" => "yup"
+          }
+
+          # Array literals work and we merge rather than overwrite:
+          default["baz"]["an_array"] = ["a", "b", "c"]
+
+          # all the same stuff works with overrides:
+
+          override["foo"] = "bar"
+
+          override["abc"]["def"]["ghi"] = "xyz"
+
+          override["baz"] = {
+            "more_nested_stuff" => "yup"
+          }
+
+          override["baz"]["an_array"] = ["a", "b", "c"]
+        EOH
+      end
+
+      let(:expected_merged_attrs) do
+        {
+          "foo" => "bar",
+          "abc" => { "def" => { "ghi" => "xyz" } },
+          "baz" => {
+            "more_nested_stuff" => "yup",
+            "an_array" => ["a", "b", "c"]
+          }
+        }
+      end
+
+      it "defines default attributes" do
+        expect(policyfile.errors).to eq([])
+        expect(policyfile.default_attributes).to eq(expected_merged_attrs)
```

Comment:
:+1: That makes sense - the 'merged' refers to the default or override attributes after finishing parsing the policyfile and all `default` or `override` DSL methods have been called.